### ISSUE TITLE
suspicious_double_ref_op doesn't give warning in generic context

### ISSUE THREAD

**Issue by safinaskar** [rust-lang/rust#112364]
### Code

```Rust
#![deny(suspicious_double_ref_op)]

fn f<T>(a: &T) {
    a.clone();
}

fn main() {
}
```

### Current output

```Shell
warning: function `f` is never used
 --> src/main.rs:3:4
  |
3 | fn f<T>(a: &T) {
  |    ^
  |
  = note: `#[warn(dead_code)]` on by default

warning: unused return value of `clone` that must be used
 --> src/main.rs:4:5
  |
4 |     a.clone();
  |     ^^^^^^^^^
  |
  = note: cloning is often expensive and is not expected to have side effects
  = note: `#[warn(unused_must_use)]` on by default
help: use `let _ = ...` to ignore the resulting value
  |
4 |     let _ = a.clone();
  |     +++++++

warning: `lii` (bin "lii") generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.25s
```

### Rationale and extra context

This is exactly situation where `suspicious_double_ref_op` should trigger. But it does not. CC @fee1-dead

rustc version: `rustc 1.72.0-nightly (e6d4725c7 2023-06-05)`

**Comment by QuineDot** [comment]
[The `noop_method_call` lint](https://github.com/rust-lang/rust/issues/83099) covers this scenario.

[Playground.](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1337c41d7df1fe5d2a630c9414130c93)
